components/base-node.js

```diff
@@ -65,7 +79,32 @@ module.exports = {
        ar.push(item);
      }
      return ar;
-   } 
+   },
+
+   updatePromise: function(def) {
+     return promiseComponent(
 
+         _.defaults(
+             def,
+             { inPorts: _.mapObject(
+                          def.inPorts,
+                          function(port, portName) {
+
+                            return _.defaults(
+                                     port,
+                                     {
+                                       ondata: function ( payload, socketIndex ) {
+                                                 this[portName] = payload;
```

Comment:
What kind of object is 'this' supposed to be bound to, at this point?  I'm trying to figure out why ondata is assigning the payload to this[portName].   Is it because states are not yet implemented?